### PR TITLE
Devices

### PR DESCRIPTION
**Pull request by MoonDweller29**

### CODE REVIEW

File: include/magma/vk/LogicalDevice.h [hunk 1]
```diff
@@ -36,8 +40,8 @@ class LogicalDevice {
     PhysicalDevice _physDevice;
     vk::Device     _device;
 
-    vk::Queue       _graphicsQueue;
-    vk::Queue       _presentQueue;
+    Queue       _graphicsQueue;
+    Queue       _presentQueue;
     vk::CommandPool _graphicsCmdPool;
```
Comment (aiosadchy) [inline include/magma/vk/LogicalDevice.h:45]: delet this

File: include/magma/vk/Window.h [hunk 2]
```diff
@@ -21,21 +21,21 @@ class Window {
     VkExtent2D getResolution() const { return {_width, _height}; }
     void       updateResolution();
 
-    GLFWwindow * getGLFWp()         { return _window;         }
-    VkSurfaceKHR getSurface() const { return _surface;        }
-    bool         wasResized() const { return _wasResized;     }
-    Keyboard *   getKeyboard()      { return _keyboard.get(); }
-    Mouse *      getMouse()         { return _mouse.get();    }
+    GLFWwindow *   getGLFWp()         { return _window;         }
```
Comment (aiosadchy) [inline include/magma/vk/Window.h:24]: What does 'p' in 'GLFW**p**' stands for?

Comment (MoonDweller29) [reply include/magma/vk/Window.h:24]: 'p' means "pointer" :D

File: src/magma/vk/physicalDevice/HardwareManager.cpp [hunk 3]
```diff
@@ -10,59 +10,39 @@ HardwareManager::HardwareManager(vk::Instance instance) :
...
-
         i++;
     }
 
-    return indices;
+    if ((requirements.graphicsSupport.isRequired() && (graphicsFamiliesCount == 0)) ||
+        (requirements.computeSupport.isRequired() && (computeFamiliesCount == 0)) ||
+        (requirements.surface.isRequired() && (presentFamiliesCount == 0)) ) {
+        return false;
+    }
```
Comment (aiosadchy) [inline src/magma/vk/physicalDevice/HardwareManager.cpp:40]: looks kinda... bad!

Comment (aiosadchy) [reply src/magma/vk/physicalDevice/HardwareManager.cpp:40]: still works though... .. .

File: include/magma/vk/Window.h (hunk 2 above)
Comment (MoonDweller29) [reply include/magma/vk/Window.h]: renamed this method to getGlfwWindow()

File: include/magma/vk/LogicalDevice.h (hunk 1 above)
Comment (MoonDweller29) [reply include/magma/vk/LogicalDevice.h]: Done

File: src/magma/vk/Window.cpp [hunk 4]
```diff
@@ -72,6 +69,11 @@ void Window::updateResolution() {
     _wasResized = false;
 }
 
+void Window::setTitle(const std::string &title) {
+    glfwSetWindowTitle(_window, title.c_str());
+}
+
+
```
Comment (Nastormo) [inline src/magma/vk/Window.cpp:76]: Please del extra line

Comment (MoonDweller29) [reply src/magma/vk/Window.cpp]: done

**Review by Nastormo — APPROVED** [review]

**Pull request merged**